nntrainer/layers/layer_impl.cpp

```diff
@@ -35,6 +36,65 @@ void LayerImpl::finalize(InitLayerContext &context) {
 
 void LayerImpl::setProperty(const std::vector<std::string> &values) {
   loadProperties(values, *layer_impl_props);
+
+  /// @todo: deprecate this in favor of loadProperties
+  for (unsigned int i = 0; i < values.size(); ++i) {
+    std::string key;
+    std::string value;
+    std::stringstream ss;
+
+    if (getKeyValue(values[i], key, value) != ML_ERROR_NONE) {
+      throw std::invalid_argument("Error parsing the property: " + values[i]);
+    }
+
+    if (value.empty()) {
+      ss << "value is empty: key: " << key << ", value: " << value;
+      throw std::invalid_argument(ss.str());
+    }
+
+    /// @note this calls derived setProperty if available
+    setProperty(key, value);
+  }
+}
+
+void LayerImpl::setProperty(const std::string &type_str,
+                            const std::string &value) {
+  int status = ML_ERROR_NONE;
+  LayerV1::PropertyType type =
+    static_cast<LayerV1::PropertyType>(parseLayerProperty(type_str));
+
+  switch (type) {
```

Comment:
Still, it is not clear, which property should go layer_node and which should go here and layer. I'm sure you will make doc but please keep in mind that there are developers who think like me.

Reply:
Added description about the LayerImpl in details section of LayerImpl header, which covers the help it provides and the properties it does. Added the similar kind of description in LayerNode.h header as well.